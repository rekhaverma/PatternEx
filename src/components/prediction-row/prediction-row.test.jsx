import React from 'react';
import PredictionRow from './prediction-row.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<PredictionRow />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      className: 'test-class',
      entity_name: 'entity_name',
      behavior: 'behavior',
      predictions: ['predictions'],
      solved: true,
      tags: ['tag-1', 'tag-2'],
      timestamp: 'timestamp',
      onInspect: jest.fn(),
      onConfirm: jest.fn(),
      onDeny: jest.fn(),
      onEntityClick: jest.fn(),
    };
    wrapper = shallow(<PredictionRow {...props} />);
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onEntityClick when element is clicked', () => {
    wrapper.find('.test-class__entity').simulate('click');
    expect(props.onEntityClick.mock.calls.length).toBe(1);
  });
});
