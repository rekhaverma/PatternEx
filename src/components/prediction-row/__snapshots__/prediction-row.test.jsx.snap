// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PredictionRow /> should match with snapshot 1`] = `
<div
  className="test-class__row"
>
  <div
    className="test-class__type"
  >
    <div
      key="tag-1_entity_name"
    >
      <Tooltip
        hideWhenHoveringTooltipContent={false}
        managerStyle={
          Object {
            "display": "flex",
          }
        }
        position="bottom"
        style={Object {}}
        trigger={
          <span
            className=""
          />
        }
      >
        <FormattedMessage
          id="predictions.type.tag-1"
          values={Object {}}
        />
      </Tooltip>
    </div>
    <div
      key="tag-2_entity_name"
    >
      <Tooltip
        hideWhenHoveringTooltipContent={false}
        managerStyle={
          Object {
            "display": "flex",
          }
        }
        position="bottom"
        style={Object {}}
        trigger={
          <span
            className=""
          />
        }
      >
        <FormattedMessage
          id="predictions.type.tag-2"
          values={Object {}}
        />
      </Tooltip>
    </div>
  </div>
  <div
    className="test-class__content"
  >
    <span
      className="test-class__entity"
      onClick={[Function]}
    >
      entity_name
    </span>
    <span
      className="test-class__timestamp"
    >
      timestamp
    </span>
    <div
      className="test-class__list"
    >
      <div
        className="test-class__item"
        key="predictions#0"
      >
        <p
          className="test-class__name"
        >
          predictions
        </p>
        <div
          className="test-class__actions +history"
        >
          <span>
            Change Label
          </span>
          <span
            className="icon-deny"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
